stock_picking_deliveryslip_subtotals/views/stock_picking_deliveryslip_subtotals.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="stock_picking_report_deliveryslip_subtotals_document" inherit_id="stock.report_delivery_document">
+
+        <xpath expr="//tr[@t-as='move_line']" position="before">
+            <t t-set="decimal_precision" t-value="request.env['decimal.precision'].precision_get('Product Unit of Measure')"/>
+            <t t-set="cont" t-value="0"/>
+        </xpath>
+
+        <xpath expr="//tr[@t-as='move_line']" position="attributes">
+            <attribute name="t-foreach">o.move_line_ids.sorted(key=lambda x: x.product_id.id)</attribute>
+        </xpath>
+        
+        <xpath expr="//tr[@t-as='move_line']" position="inside">
+            <t t-set="cont" t-value="cont+1"/>
+            <t t-if="(len(o.move_line_ids.filtered(lambda x: x.product_id.id == move_line.move_id.product_id.id)) == cont)">
+                <tr>
+                    <td></td>
+                    <td class="border-top border-dark text-center">
+                    <strong style="color: #859b29;">Subtotal</strong>
+                    </td>
+                    <td class="border-top border-dark text-center">
+                    <span t-esc="sum(o.move_line_ids.filtered(lambda x: x.product_id.id == move_line.move_id.product_id.id).mapped('qty_done'))" t-options="{'widget': 'float', 'precision': decimal_precision}"/>
```

Comment:
This is inefficient (every line calculates aggregated data that could be obtained once). I'll take a look and try to provide you a better solution